atc/db/team.go

```diff
@@ -431,22 +430,25 @@ func savePipeline(
 			"instance_vars":   instanceVars,
 		}
 		var ordering sql.NullInt64
-		err := psql.Select("max(ordering)").
+		var secondaryOrdering sql.NullInt64
+		err := psql.Select("max(ordering), max(secondary_ordering)").
 			From("pipelines").
 			Where(sq.Eq{
 				"team_id": teamID,
 				"name":    pipelineRef.Name,
 			}).
 			RunWith(tx).
 			QueryRow().
-			Scan(&ordering)
+			Scan(&ordering, &secondaryOrdering)
 		if err != nil {
 			return 0, false, err
 		}
 		if ordering.Valid {
 			values["ordering"] = ordering.Int64
+			values["secondary_ordering"] = secondaryOrdering.Int64 + 1
 		} else {
 			values["ordering"] = sq.Expr("currval('pipelines_id_seq')")
+			values["secondary_ordering"] = 1
 		}
```

Comment:
Not really, we just need a consistent ordering here and it felt more natural that way, Also Ordering is not 0-base at the beginning it only change his ordering to 0-base after re-ordering some pipelines.

Reply:
To clarify the second part - we just take `currval('pipelines_id_seq')`, which will leave you with some value of `ordering` that is larger than any other pipeline currently.

`secondary_ordering` on the other hand is easy to keep as 1-based, and felt more natural to do so (since the `row_number` function we use in the migration is also 1-based, and it'd be a bit awkward to subtract 1 when all you really care about is the relative ordering).

That said, there isn't really a good reason for it - we're happy to reconsider this, I don't think either @EstebanFS or I feel particularly strongly about it one way or another

Reply:
I don't particularly care either, I guess it's just odd that what is (essentially) a list starting at `1`